### PR TITLE
added k switch for insecure certs

### PR DESCRIPTION
**Pull request by silv3rr**
Hi @SReject,

Thanks for JFM, i've used it in 2 scripts now and works nicely. I also like how in the new versions the JS part is formatted in seperate file.

For an API on localhost with selfsigned cert I got a (expected) cert error "The certificate authority is invalid or incorrect".
I could not find an option to disable cert checking so a I added a "k" switch to JSONOpen (like curl -k).

The Tests script passed and I've added tests for -k without -u" switch and 2 for JSONOpen and a known bad cert with/without "-k".

### CODE REVIEW

**Comment by SReject** [comment]
Merge latest upstream please

File: test/JSON for mIRC - Tests.mrc [hunk 1]
```diff
@@ -1145,15 +1158,37 @@ alias -l jfm_test {
...
+  JSONClose jfm_test
+
   ;; Attempt to retrieve data from a remote source
   inc %testnum
   JSONOpen -u jfm_test http://echo.jsontest.com/key/value
   if ($null !== $JSONError) {
+    if ($JSONError == INVALID_JSON) {
+      echo -ast There might be an issue with jsontest.com : $v2
+      echo -ast Possible reason could be 'Over Quota', halting further tests...
+      halt
```
Comment (SReject) [inline test/JSON for mIRC - Tests.mrc]: Don't halt, instead return the warning as though its an error

File: src/JSON For mIRC.mrc [hunk 2]
```diff
@@ -77,6 +77,7 @@ menu @SReject/JSONForMirc/Log {
 ;;     -d:  Closes the handler after the script finishes
 ;;     -b:  The input is a bvar
 ;;     -f:  The input is a file
+;;     -k:  Used with -u; Ignore all certificate errors
```
Comment (SReject) [inline src/JSON For mIRC.mrc]: use `-i` (for insecure) over `-k`

File: src/JSON For mIRC.js [hunk 3]
```diff
@@ -493,7 +502,8 @@
 
     root.JSONInstance = JSONInstance;
 
-    root.JSONCreate = function(type, source, parse) {
+    // (slv) Added: 'insecure' bool
+    JSONCreate = function(type, source, parse, insecure) {
```
Comment (SReject) [inline src/JSON For mIRC.js]: Not sure how this occured, but this should be `root.JSONCreate = ...`

**Review by SReject — CHANGES_REQUESTED** [review]
Along with these:

Merge upstream, giving upstream precedence
Bump build version `1.1.xxxx`

**Comment by silv3rr** [comment]
Applied required changes and fixed merge issues

**Review by SReject — APPROVED** [review]

**Comment by SReject** [comment]
Merged, thanks for the help!

**Pull request merged**